src/runtime_src/core/pcie/emulation/cpu_em/generic_pcie_hal2/shim.cxx

```diff
@@ -738,7 +749,12 @@ namespace xclcpuemhal2 {
       if ( mLogStream.is_open() ) {
         verbose = true;
       }
-
+      setDeviceProcessStarted(true);
```

Comment:
1.  The std::memory_order is not default sequential, so there is no guarantee the thread will ever see the value of the unsynchronized variable set on line 752.
2. It is very awkward coding to separate the control variable from the starting of the thread.  You nicely encapsulate closemessagethread, why are you encapsulating startmessagethread and manage this synchronization variable within these start and stop routines?